app/views/PillarAnalysis/AnalyticalStatementInput/StoryAnalysisModal/index.tsx

```diff
@@ -473,7 +599,38 @@ function StoryAnalysisModal(props: Props) {
                             <MarkdownEditor
                                 className={styles.editor}
                                 labelContainerClassName={styles.labelContainer}
-                                label="Information Gaps"
+                                label={(
+                                    <>
+                                        <div>
+                                            Information Gap
+                                        </div>
+                                        <div className={styles.labelContainerAction}>
+                                            <QuickActionConfirmButton
+                                                name={undefined}
+                                                title="Auto generate information gap using NLP"
+                                                message="You are about to auto generate information gap using NLP. This will replace the current information gap. Are you sure you want to continue?"
+                                                onConfirm={informationGapOnClick}
+                                                disabled={project?.isPrivate || pending}
+                                                variant="nlp-primary"
+                                            >
+                                                <Svg
+                                                    className={styles.brainIcon}
+                                                    src={brainIcon}
+                                                />
+                                            </QuickActionConfirmButton>
+                                            {project?.isPrivate && (
+                                                <div className={styles.info}>
+                                                    <IoInformation />
+                                                    <Tooltip>
+                                                        Auto generate is not available
+                                                        for private projects to
+                                                        maintain document privacy.
+                                                    </Tooltip>
+                                                </div>
+                                            )}
+                                        </div>
+                                    </>
+                                )}
```

Comment:
You  can use Header component for easy placement. Not super necessary but recommended.